### DISCUSSION TITLE
Exclude a step from progress based on some condition

### DISCUSSION THREAD

**Discussion by MohmmedAshraf** [spatie/laravel-onboard#2] [Ideas]
Hi there, Great package, as usual, thank you very much Spatie ❤️, 
I would suggest something am working on a project that requires onboarding but based on the enabled modules, 
each module has its own steps, so what I am looking for if there is an option to exclude a step from preventing the onboarding progress from being incomplete, something like this:

```php
Onboard::addStep('Step 1')
    ...
    ->excludeIf(function (Business $business) {
        return $business->hasFeature('offers');
    });
```

**Update**
I've already contributed this feature, see #3

**Comment by RhysLees** [comment]
Should be able to do this with the following:

```php
Onboard::addStep('Step 1')
    ...
    ->excludeIf(function ($model) {
        return $model->hasFeature('offers');
    });
```
